packages/core/src/sync-realtime/index.ts

```diff
@@ -321,6 +372,30 @@ export const createRealtimeSync = (
       (lb) => hexToNumber(lb.number) < hexToNumber(block.number),
     );
 
+    // recompute `unfinalizedChildAddresses`
+    unfinalizedChildAddresses.clear();
+
+    for (const filter of factories) {
+      unfinalizedChildAddresses.set(filter, new Set());
+      for (const { hash } of localChain) {
+        const factoryLogs = factoryLogsPerBlock.get(hash);
+        if (factoryLogs !== undefined) {
+          for (const log of factoryLogs) {
+            if (isLogFactoryMatched({ filter, log })) {
+              unfinalizedChildAddresses
+                .get(filter)!
+                .add(getChildAddress({ log, factory: filter }));
+            }
+          }
+        }
+      }
+    }
+
+    // delete reorged blocks from `factoryLogsPerBlock`
+    for (const { hash } of reorgedBlocks) {
+      factoryLogsPerBlock.delete(hash);
+    }
```

Comment:
Should this happen before we recompute `unfinalizedChildAddresses` ? It shouldn't matter, just double checking.